groups/bdl/bdlb/bdlb_pcgrandomgenerator.h

```diff
@@ -129,29 +129,36 @@ BSLS_IDENT("$Id: $")
 //     assert(d2 == d1);
 // }
 //..
-///Example 2: Generating a random 32-bit number
-///- - - - - - - - - - - - - - - - - - - - - - -
-// The PcgRandomGenerator constructor takes two 64-bit constants (the initial
-// state, and the rng sequence selector.  Instances of 'PcgRandomGenerator' with
-// different sequence selectors will never have random sequences that coincide.
-// In order to obtain nondeterministic output for each instance, on a Unix
-// system random bytes can be obtained from '/dev/urandom'.  The component 
-// bdlb_randomdevice provides such functionality.  Also illustrated below is a 
-// a backup mechanism for obtaining random bytes.
+///Example 2: Generating unique random sequences of 32-bit numbers
+///- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+// This example shows how one might construct two sequences of random numbers
```

Comment:
I took a stab at this example - please see the differences in the delivery style (also note that code is indented with 2 spaces after `//`):

```
///Example 2: Using stream selector to guarantee uncorrelated sequences
/// - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
//
// This example illustrates how the stream selector can be used to guarantee
// that two distinct instances of 'PCGRandomGenerator' produce uncorrelated
// sequences.
//
// First, we use the 'RandomDevice' to choose the initial states for the
// generators using a source of true randomness:
//..
//  uint64_t state1;
//  int rc = bdlb::RandomDevice::getRandomBytes(
//      reinterpret_cast<unsigned char *>(&state1), sizeof(state1));
//  (void)rc;  // error handling omitted
//
//  uint64_t state2;
//  int rc = bdlb::RandomDevice::getRandomBytes(
//      reinterpret_cast<unsigned char *>(&state2), sizeof(state2));
//  (void)rc;  // error handling omitted
//..
// Then we select two distinct stream selectors for the generators, which, due
// to the PCG algorithmic properties will guarantee that the sequences will be
// uncorrelated even if the initial state is exactly the same:
//..
//  const uint64_t streamSelector1 = 1;
//  const uint64_t streamSelector2 = 2;
//..
// Finally, we initialize the generators with their respective initial states
// and stream selectors and check that they produce distinct sequences of
// random numbers.  The check is guaranteed to pass even in the rare, but
// possible case of 'state1 == state2'.
//..
//  bdlb::PcgRandomGenerator rng1(state1, streamSelector1);
//  bdlb::PcgRandomGenerator rng2(state2, streamSelector2);
//
//  const int NUM_VALUES = 1000;
//  bsl::vector<bsl::uint32_t> sequence1(NUM_VALUES);
//  bsl::vector<bsl::uint32_t> sequence2(NUM_VALUES);
//
//  for (int i = 0; i < NUM_VALUES; ++i) {
//      sequence1[i] = rng1.generate();
//      sequence2[i] = rng2.generate();
//  }
//
//  assert(sequence1 != sequence2);
//..
```